### ISSUE TITLE
[List/List Item] Mitigate action access when toggling the `disabled` state

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-design-system#7335]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Summary

Ensure all List Items and their actions are available to users when toggling the `disabled` states.

![a11y-listitem-disabled-actions](https://github.com/Esri/calcite-design-system/assets/5023024/b3b2f056-a0dd-4f34-beea-6fcd8b0e3232)

### Actual Behavior

When List Items `disabled` state is toggled, its actions remain inaccessible to keyboard users, while available to mouse users.

### Expected Behavior

Either no `tabindex` should be present via the interactive helper, or the `tabindex` should be updated when changing between the `disabled` state. [Additional context provided via Matt](https://github.com/Esri/calcite-design-system/issues/7254#issuecomment-1629396533) addressed in #7254.

### Reproduction Sample

https://codepen.io/geospatialem/pen/XWyzpXX

### Reproduction Steps

1. Open the [sample](https://codepen.io/geospatialem/pen/XWyzpXX)
2. Select the first item in the list by pressing the `Enter` key
3. Press the `Enter` key again to remove any disabled List items
4. Press `Tab` to navigate to the apple's action
5. Press `Tab` again and observe the next item is the mango's action

### Reproduction Version

1.5.0-next.17

### Relevant Info

https://github.com/Esri/calcite-design-system/issues/7254#issuecomment-1629396533

### Regression?

N/A

### Priority impact

p2 - want for current milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react

### Esri team

Calcite (dev)

**Comment by jcfranco** [comment]
Found out the culprit. The item's host focus behavior is being [set as `managed`](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/list-item/list-item.tsx#L316C34-L316C41), but the item's `tabindex` is not being set by the parent (e.g., pick/value list-item's [`tabindex`](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/value-list-item/value-list-item.tsx#L155) is managed by [its](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/value-list/value-list.tsx#L355) [parent](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/pick-list/shared-list-logic.ts#L268) [list](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/pick-list/shared-list-logic.ts#L223-L225)). This should be a low-effort fix and feasible for this milestone. I'll also update `updateHostInteraction`'s doc to give context on its `hostIsTabbable` argument).

**Comment by geospatialem** [comment]
Verified in `1.10.0-next.7` with https://codepen.io/geospatialem/pen/XWyzpXX.
